server/controllers/chartsDataController/index.js

```diff
@@ -6,18 +6,17 @@ import SubjectModel from '../../models/SubjectModel'
 import BarChartService from '../../services/BarChartService'
 import BarChartTableService from '../../services/BarChartTableService'
 import CsvService from '../../services/CSVService'
-import FiltersModel from '../../models/FiltersModel'
+import StudiesModel from '../../models/StudiesModel'
+import FiltersService from '../../services/FiltersService'
 
 const show = async (req, res, next) => {
   try {
     const { dataDb, appDb } = req.app.locals
+    const userSites = StudiesModel.sanitizeAndSort(req.session.userAccess)
     const { chart_id } = req.params
-    const { userAccess } = req.session
     const parsedQueryParams = qs.parse(req.query)
-    const filters = FiltersModel.calculateFilters(
-      parsedQueryParams.filters,
-      userAccess
-    )
+    const filtersService = new FiltersService(parsedQueryParams.filters)
+    const filters = filtersService.sanitize(userSites)
```

Comment:
Do we want to instantiate the services here and then again in the SubjectModel.allForAssessment..?

Reply:
hmm yeah I'm not sure. I could go either way. What do you think about this if we want to only call it once? https://github.com/AMP-SCZ/dpdash/pull/442/commits/6e1f8aacbbc8f2c3284178dbb915a7578c97c26e

Reply:
That's what I was struggling with for a while. Is passing a service in that manner a good pattern